packages/flutter/test/material/text_field_test.dart

```diff
@@ -1018,6 +1018,82 @@ void main() {
     expect(controller.text, 'abc d${testValue}ef ghi');
   });
 
+  testWidgets(
+    'Check the toolbar appears below the TextField when there is not enough space above the TextField to show it',
```

Comment:
This seems like it's from your other PR (https://github.com/flutter/flutter/pull/29809) and it verifies the top/bottom position, but you should write a test that verifies the arrow position.

I think if you just cover the 3 cases it should be fine.  Test that selecting in the middle of an input puts the arrow in the middle (as before), selecting the first word puts the arrow left of center, and selecting the last word puts the arrow right of center.

It might be hard to find the arrow in a test, but I think you can do it.  Here is a snippet from another test that finds something by name and descendant, it should help here:

```dart
Finder findFadeUpwardsPageTransition() {
  return find.descendant(
    of: find.byType(MaterialApp),
    matching: find.byWidgetPredicate((Widget w) => '${w.runtimeType}' == '_FadeUpwardsPageTransition'),
  );
}
```